### ISSUE TITLE
Atom and Module is considered a type, why?

### ISSUE THREAD

**Issue by manuel-rubio** [ExpressApp/construct#38]
https://github.com/ExpressApp/construct/blob/301df7388b9283f39ee43bb209c06f3aa055c8e7/lib/construct/compiler/ast/ast_types.ex#L158

I'm defining this:

```elixir
    field(:module, Hemdal.Config.Module, default: Hemdal.Host.Local)
```

And it's telling me that `Hemdal.Host.Local` must be a type, why? I was defining the type as `Hemdal.Config.Module.t()` which is (by the way) `module()`, why is it needed to add `Hemdal.Host.Local.t()` as well?

And even if I have:

```elixir
    field(:status, Status, default: :ok)
```

Why the type for the default value is defined as `ok.t()` ???

**Comment by artemeff** [comment]
сonstruct was originally made for user data and evolved for specific needs, while remaining a general-purpose utility.

I never thought of using it for validating the configuration inside the system, our cases is for validating external data, comes from users input.

you are using a custom type and the construct compiler detects it as a custom type and works with default values as a custom type. that's why I released 3.0.0-rc.0 - just to be sure that this deep refactoring not broke others code and works as expected.

your case is interesting, I'll see what I can do :)

**Comment by artemeff** [comment]
@manuel-rubio fixed a case when atoms always treated as custom types, so now I check if it's a really construct module, otherwise just use an atom.

I want to leave typespec generation a simple as possible, so for your case I made another feature — you can provide your own typespec
